src/msftoolbox/gdelt/articles.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime
+import requests
+from urllib.error import HTTPError
+import newspaper
+
+class GDELTExtractor:
+    def __init__(
+        self,
+        sort="HybridRel",
+        limit=50,
+        mode="ArtList"
+    ):
+        self.sort = sort
+        self.mode = mode
+        self.limit = limit
+
+    def format_date(self, date) -> str:
+        date = datetime.strptime(date, "%Y-%m-%d")
+        return datetime.strftime(date, "%Y%m%d%H%M%S")
+
+    def list_reports(
+        self,
+        start_date: str,
+        end_date: str,
+        query_value: str,
+        country_filter: list = None,
+        response_format: str = "JSON"
+    ) -> list:
+        base_url = "https://api.gdeltproject.org/api/v2/doc/doc"
+        start_date = self.format_date(start_date)
+        end_date = self.format_date(end_date)
+
+        params = {
+            "query": f"{query_value} AND {country_filter}",
+            "mode": self.mode,
+            "startdatetime": start_date,
+            "enddatetime": end_date,
+            "format": response_format,
+            "trans": "googtrans"
+        }
+
+        response = requests.get(base_url, params=params)
+        if response.status_code != 200:
+            raise HTTPError(f"HTTP error: {response.status_code}")
+
+        response_data = response.json()["articles"]
```

Comment:
I am thinking that if we allow for any other response_format than JSON, this will break, so we should probably pin it